src/connectionhealthpolicy/VideoEncodingCpuConnectionHealthPolicy.ts

```diff
@@ -12,19 +12,23 @@ export default class VideoEncodingCpuConnectionHealthPolicy
   implements ConnectionHealthPolicy {
   private readonly consecutiveHighEncodeCpuThreshold: number;
   private readonly highEncodeCpuMsThreshold: number;
+  private readonly highEncodeCpuMsPerFrameThreshold: number;
   private consecutiveHighEncodeCpuCnt = 0;
 
   constructor(configuration: ConnectionHealthPolicyConfiguration, data: ConnectionHealthData) {
     super(configuration, data, VideoEncodingConnectionHealthPolicyName.VideoEncodingCpuHealth);
     this.consecutiveHighEncodeCpuThreshold = configuration.consecutiveHighEncodeCpuThreshold;
     this.highEncodeCpuMsThreshold = configuration.highEncodeCpuMsThreshold;
+    this.highEncodeCpuMsPerFrameThreshold = configuration.highEncodeCpuMsPerFrameThreshold;
   }
 
   health(): number {
+    const videoEncodingTimeIsHigh =
+      this.currentData.videoEncodingTimeInMs >= this.highEncodeCpuMsThreshold &&
```

Comment:
It _seems_ like we don't need both checks. But I assume you have a good reason already? :)

If so, maybe it can be a comment

Reply:
Having both checks there is to avoid degrading codec in two conditions:
1. When framerate is low but per-frame encoding time is high (e.g., high resolution screen share with low motion on screen, encoder sometimes just encode like one frame every second. There's no need to degrade codec if each of these frames takes 50 ms to encode);
2. When frame rate is high (e.g., high framerate and SVC at the same time).

I don't think these two conditions can be excluded with the same check. So kept both here.

Reply:
Added a comment to explain the logic